### ISSUE TITLE
GetPath doesn't like barred polys mixed with total and nearest polys

### ISSUE THREAD

**Issue by Kawa-oneechan** [Kawa-oneechan/SCI11-Plus#7]
Results are somewhat disastrous.

![image](https://user-images.githubusercontent.com/12450280/141207955-270fe5da-46dc-4694-9a51-077310a48551.png)

Starting from the left, clicking near the barred poly works fine but clicking further down the right half of the screen makes the pathfinder ignore the barred poly for that request for some reason. This somehow results in memory issues when changing rooms.

Some experimentation reveals that dummying out the total _and_ nearest polys lets the barred poly always be considered, and nothing bad ever happens. Which explains why I never triggered this in _The Dating Pool_ -- that only uses contained and barred polys.

SCI11+ is based on SCI 1.001.100. Thirteen builds later in _Leisure Suit Larry 6_ this bug was already corrected, and apparently it wasn't there in earlier builds either.

**Comment by pandapawskat** [comment]
Adding the update that we discovered the issue isn't related to poly type, it seems to be triggered by the route crossing multiple polys of any type. So this can be repro'ed by making the total and nearest polys barred as well.

It feels like the issue comes from the code getting the order the polys are intersected wrong and immediately routing to the furthest one instead of the nearest, but code inspection doesn't find an obvious flaw.

I have been playing with a new routing algorithm based on the pretty standard visibility graph technique used in robot motion planning, but it has a ways to go before I would consider it reliable in various edge cases.

I've also pulled out the original pathing logic into a standalone VS project so I can experiment with it in a debugger. Unfortunately I haven't had the time to play with it more yet.
